packages/block-editor/src/components/inserter/block-list.js

```diff
@@ -127,16 +114,19 @@ export function InserterBlockList( {
 	}, [ filterValue, debouncedSpeak ] );
 
 	const hasItems = ! isEmpty( filteredItems );
-	const hasChildItems = childItems.length > 0;
 
 	return (
 		<div>
-			<ChildBlocks
-				rootClientId={ rootClientId }
-				items={ childItems }
-				onSelect={ onSelectItem }
-				onHover={ onHover }
-			/>
+			{ hasChildItems && (
+				<ChildBlocks
+					rootClientId={ rootClientId }
+					// Pass along every block, as getInserterItems() will have
+					// already filtered out non-child blocks.
```

Comment:
This comment makes me wonder what's special about `ChildBlocks` component. Why not just render `BlockTypesList` directly?

Reply:
They receive different visual treatments.

Here's `ChildBlocks`:

<img width="365" alt="Screen Shot 2020-06-18 at 10 35 22" src="https://user-images.githubusercontent.com/612155/84964821-7c5a1580-b150-11ea-90bb-ee380ccf6a8c.png">

Here's `InserterPanel`:

<img width="366" alt="Screen Shot 2020-06-18 at 10 41 11" src="https://user-images.githubusercontent.com/612155/84964835-8aa83180-b150-11ea-9813-363cd1d0ec3e.png">

Perhaps we should unify the two? I'd rather address this as a separate issue, though.

In the meantime I agree that `ChildBlocks` should at least work the same as `InserterPanel` where the parent is responsible for passing through `BlockTypes`. I've changed that in 866b98e2f4.

Reply:
Yes, seems like unnecessary complexity for now. I believe the visual difference is from the old design.